### PR TITLE
feat(mobile): Added React Native Paper component lib and themes

### PR DESCRIPTION
**Pull request by jchiversAND**
Added React Native Paper component library for components, and custom themes for Paper and
Navigation components, along with brand colors.

Separated into their own files for customisability.

Actual values for colours and themes to be decided on/finalised.
[[Dev] Colour Theming](https://app.gitkraken.com/glo/card/0bf6d83f6c5347bf9e77be90036b9072)

### CODE REVIEW

File: mobile/styles/ColorTheme.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  /* Red */
+  red: '#EE5F5E',
+  redLight: '#F6B7B6',
+  redDark: '#941010',
+
+  /* Green */
+  green: '#84DD63',
+  greenLight: '#D1F3C5',
+  greenDark: '#499E2A',
+};
```
Comment (Ehuf) [inline mobile/styles/ColorTheme.js:21]: Probably want to include whites greys and blacks as well

File: mobile/styles/PaperTheme.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+const PaperTheme = {
+  dark: false,
+  roundness: 4,
+  colors: {
+    primary: '#0D749E',
```
Comment (Ehuf) [inline mobile/styles/PaperTheme.js:5]: Are we not able to import these hex values from a central source? looks like theres a few sources of truth

File: mobile/yarn.lock [hunk 3]
```diff
@@ -969,6 +969,14 @@
   resolved "https://registry.yarnpkg.com/@bcoe/v8-coverage/-/v8-coverage-0.2.3.tgz#75a2e8b51cb758a7553d6804a5932d7aace75c39"
```
Comment (Ehuf) [inline mobile/yarn.lock:969]: Probably don't want to include this file

File: mobile/src/screens/Home/Home.jsx [hunk 4]
```diff
@@ -1,27 +1,34 @@
...
 const Title = styled.Text`
-    font-size: 20px;
-    font-weight: 501;
-    color: palevioletred;
+  font-size: 20px;
+  font-weight: 500;
+  color: palevioletred;
+`;
+
+const StyledButton = styled(Button)`
```
Comment (adamjamesturner93) [inline mobile/src/screens/Home/Home.jsx:19]: Can we move things like Title and Button into a styles directory? Two main benefits:

- All styles for a given button/title etc are managed in one place
- Decouples Screens from things like React Native Paper and emotion.  If we want to change this, we only need to change the individual component and the app will follow suit

https://medium.com/javascript-in-plain-english/javascript-design-patterns-adapter-explained-cbcffbb4b8bc

File: mobile/styles/PaperTheme.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    text: '#000000',
+    onBackground: '#000000',
+    onSurface: '#000000',
+    disabled: 'rgba(0, 0, 0, 0.26)',
+    placeholder: 'rgba(0, 0, 0, 0.54)',
+    backdrop: 'rgba(0, 0, 0, 0.5)',
+    notification: '#f50057',
+  },
+  fonts: {
+    regular: { fontFamily: 'Roboto, "Helvetica Neue", Helvetica, Arial, sans-serif', fontWeight: '400' },
```
Comment (adamjamesturner93) [inline mobile/styles/PaperTheme.js:19]: Just trusting these match the Figma! 😛

File: mobile/yarn.lock (hunk 3 above)
Comment (adamjamesturner93) [reply mobile/yarn.lock:969]: We do :)

File: mobile/styles/PaperTheme.js (hunk 2 above)
Comment (jchiversAND) [reply mobile/styles/PaperTheme.js:5]: Oops yep was meant to update the theme files, on it

File: mobile/styles/ColorTheme.js (hunk 1 above)
Comment (jchiversAND) [reply mobile/styles/ColorTheme.js:40]: Added :)

File: mobile/styles/PaperTheme.js (hunk 2 above)
Comment (jchiversAND) [reply mobile/styles/PaperTheme.js]: Done :)

File: mobile/styles/PaperTheme.js (hunk 5 above)
Comment (jchiversAND) [reply mobile/styles/PaperTheme.js]: Fixed :)

File: mobile/src/screens/Home/Home.jsx (hunk 4 above)
Comment (jchiversAND) [reply mobile/src/screens/Home/Home.jsx]: Done :)

File: mobile/src/screens/Login/Login.jsx [hunk 6]
```diff
@@ -1,16 +1,17 @@
...
-        <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
-            <Text>Login View Screen</Text>
-            <Button title="Go to Login View... again" onPress={() => navigation.push('Login')} />
-            <Button title="Go to Home" onPress={() => navigation.navigate('Home')} />
-            <Button title="Go back" onPress={() => navigation.goBack()} />
-            <Button title="Go back to first screen in stack" onPress={() => navigation.popToTop()} />
-        </View>
-    );
+  return (
+    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
```
Comment (adamjamesturner93) [inline mobile/src/screens/Login/Login.jsx:7]: Let's avoid inline styles - their horrendous for consistency

File: mobile/src/screens/Home/Home.jsx [hunk 7]
```diff
@@ -1,27 +1,24 @@
 import React from 'react';
-import { Button } from 'react-native';
+import { Button, Title } from '../../../styles/components';
 import styled from '@emotion/native';
+import colors from '../../../styles/constants/ColorScheme';
 
 const Container = styled.View`
```
Comment (adamjamesturner93) [inline mobile/src/screens/Home/Home.jsx:6]: Can the Container in here be abstracted? Removes @emotion entirely from the file.

File: mobile/src/screens/Home/Home.jsx [hunk 8]
```diff
@@ -1,27 +1,24 @@
...
-    return (
-        <Container>
-            <Title>React Native with 👩‍🎤 Emotion</Title>
-            <Button title="Login" onPress={() => navigation.navigate('Login')} />
-        </Container>
-    );
+  return (
+    <Container>
+      <Title>React Native with 👩‍🎤 Emotion</Title>
+      <Button color={colors.yellow} onPress={() => navigation.navigate('Login')}>
```
Comment (adamjamesturner93) [inline mobile/src/screens/Home/Home.jsx:17]: Can we but have "primary, secondary" button types? Happy, for now, to make it arbitrary until we have designs provided.  But sets a nice precedent.

File: mobile/src/screens/Home/Home.jsx (hunk 7 above)
Comment (jchiversAND) [reply mobile/src/screens/Home/Home.jsx:6]: Deleted for now as was just placeholder :)

File: mobile/src/screens/Login/Login.jsx (hunk 6 above)
Comment (jchiversAND) [reply mobile/src/screens/Login/Login.jsx:7]: Destroyed :)

**Comment by adamjamesturner93** [comment]
Looks good 👍 
When we make the screens for real, we will want to move the `<Text>` to being styled text fields. But I think this is an excellent base point

**Pull request merged**